R/ff_run.R

```diff
@@ -69,64 +69,265 @@ ff_run <- function(shape = NULL, country = NULL, prediction_dates = NULL,
+                   certainty_threshold = 0.5,
                    filter_features = "initialforestcover",
                    filter_conditions = ">0",
-                   accuracy_csv = NULL,
-                   importance_csv = NA,
+                   accuracy_output_path = NULL,
+                   importance_output_path = NA,
                    verbose = TRUE,
                    autoscale_sample = FALSE,
-                   validation = FALSE) {
+                   validation = FALSE,
+                   groundtruth_pattern = Sys.getenv("DEFAULT_GROUNDTRUTH")) {
   fixed_sample_size <- 6e6
-  sample_size <- 0.3
 
-  if (!has_value(shape) && !has_value(country)) {
-    stop("either input shape or country should be given")
+  corrected_date_input <- check_dates(
+    train_dates, validation_dates,
+    prediction_dates, validation, pretrained_model_path, groundtruth_pattern
+  )
+  validation <- corrected_date_input$validation
+  prediction_dates <- corrected_date_input$prediction_dates
+  pretrained_model_path <- corrected_date_input$pretrained_model_path
+  train_dates <- corrected_date_input$train_dates
+
+  shape_and_tiles <- check_folder_and_input(
+    ff_folder, country, shape, train_dates,
+    prediction_dates, model_save_path, predictions_save_path
+  )
+  shape <- shape_and_tiles$shape
+  tiles <- shape_and_tiles$tiles
+
+  # Train model if not provided
+  if (is.null(pretrained_model_path)) {
+    sample_fraction <- determine_sample_fraction(
+      autoscale_sample, ff_folder,
+      shape, train_dates,
+      filter_conditions, filter_features,
+      groundtruth_pattern,
+      ff_prep_parameters,
+      validation, fixed_sample_size, verbose
+    )
+    ff_cat("Preparing data\nLooking in folder", ff_folder, verbose = verbose, color = "green")
+
+    data_and_parameters <- prepare_training_data(
+      ff_folder, shape, train_dates, filter_conditions, filter_features,
+      sample_fraction, groundtruth_pattern, validation,
+      ff_prep_parameters, verbose
+    )
+    train_input_data <- data_and_parameters$train_input_data
+    ff_prep_params_combined <- data_and_parameters$ff_prep_params_combined
+
+    train_input_data <- prepare_validation_data(
+      train_input_data,
+      train_dates, validation_dates, ff_prep_params_combined, verbose
+    )
+
+    pretrained_model_path <- prepare_and_train(
+      train_input_data, model_save_path,
+      validation, validation_dates, ff_train_parameters, verbose
+    )
   }
-  if (has_value(shape)) {
-    ForestForesight::check_spatvector(shape,
-      check_size = has_value(train_dates)
+  if (has_value(importance_output_path)) {
+    if (has_value(model_save_path)) {
+      ff_importance(model = model_save_path, output_csv = importance_output_path, append = TRUE)
+    } else {
+      ff_importance(model = pretrained_model_path, output_csv = importance_output_path, append = TRUE)
+    }
+  }
+  # Predict
+  if (prediction_dates[1] == "3000-01-01") {
```

Comment:
maybe this whole chunk can be a separate function? we had a similar thing with for loops of tiles in `ff_prep`